pkg/server/plugin/nodeattestor/aws/iid_test.go

```diff
@@ -0,0 +1,424 @@
+				},
+			},
+		},
+	}
+}
+
+func (s *IIDAttestorSuite) TestClientAndIDReturns() {
+	zeroDeviceIndex := int64(0)
+	nonzeroDeviceIndex := int64(1)
+	instanceStoreType := ec2.DeviceTypeInstanceStore
+
+	originalAWSPublicKey := s.plugin.config.awsCaCertPublicKey
+	defer func() {
+		s.plugin.config.awsCaCertPublicKey = originalAWSPublicKey
+	}()
+	s.plugin.config.awsCaCertPublicKey = &s.rsaKey.PublicKey
+
+	tests := []struct {
+		desc                string
+		mockExpect          func(mock *mock_aws.MockEC2Client)
+		expectID            string
+		expectErr           string
+		replacementTemplate *template.Template
+	}{
+		{
+			desc: "error on call",
+			mockExpect: func(mock *mock_aws.MockEC2Client) {
+				mock.EXPECT().DescribeInstances(&ec2.DescribeInstancesInput{
+					InstanceIds: []*string{&testInstance},
+				}).Return(nil, errors.New("client error"))
+			},
+			expectErr: "client error",
+		},
+		{
+			desc: "non-zero device index",
+			mockExpect: func(mock *mock_aws.MockEC2Client) {
+				output := getDefaultDescribeInstancesOutput()
+				output.Reservations[0].Instances[0].RootDeviceType = &instanceStoreType
+				output.Reservations[0].Instances[0].NetworkInterfaces[0].Attachment.DeviceIndex = &nonzeroDeviceIndex
+				mock.EXPECT().DescribeInstances(&ec2.DescribeInstancesInput{
+					InstanceIds: []*string{&testInstance},
+				}).Return(&output, nil)
+			},
+			expectErr: "verifying the EC2 instance's NetworkInterface[0].DeviceIndex is 0",
+		},
+		{
+			desc: "success, default template",
+			mockExpect: func(mock *mock_aws.MockEC2Client) {
+				output := getDefaultDescribeInstancesOutput()
+				output.Reservations[0].Instances[0].RootDeviceType = &instanceStoreType
+				output.Reservations[0].Instances[0].NetworkInterfaces[0].Attachment.DeviceIndex = &zeroDeviceIndex
+				mock.EXPECT().DescribeInstances(&ec2.DescribeInstancesInput{
+					InstanceIds: []*string{&testInstance},
+				}).Return(&output, nil)
+			},
+			expectID: "spiffe://example.org/spire/agent/aws_iid/test-account/test-region/test-instance",
+		},
+		{
+			desc: "success, different template",
+			mockExpect: func(mock *mock_aws.MockEC2Client) {
+				output := getDefaultDescribeInstancesOutput()
+				output.Reservations[0].Instances[0].RootDeviceType = &instanceStoreType
+				output.Reservations[0].Instances[0].NetworkInterfaces[0].Attachment.DeviceIndex = &zeroDeviceIndex
+				mock.EXPECT().DescribeInstances(&ec2.DescribeInstancesInput{
+					InstanceIds: []*string{&testInstance},
+				}).Return(&output, nil)
+			},
+			replacementTemplate: template.Must(template.New("agent-svid").Parse("{{ .PluginName}}/{{ .Region }}/{{ .AccountID }}/{{ .InstanceID }}")),
+			expectID:            "spiffe://example.org/spire/agent/aws_iid/test-region/test-account/test-instance",
+		},
+	}
+
+	for _, tt := range tests {
+		s.T().Run(tt.desc, func(t *testing.T) {
+			mockCtl := gomock.NewController(s.T())
+			defer mockCtl.Finish()
+
+			ec2Client := mock_aws.NewMockEC2Client(mockCtl)
```

Comment:
seems like a lot of this scaffolding could be simpler if 1) instantiated a new plugin for each test case instead of using the one on the suite struct, 2) used the Configure() call to set up the template, "skip" settings, etc.